packages/editor-ui/src/composables/usePinnedData.ts

```diff
@@ -85,14 +85,16 @@ export function usePinnedData(
 		if (!nodeType || (checkDataEmpty && dataToPin.length === 0)) return false;
 
 		const workflow = workflowsStore.getCurrentWorkflow();
-		const outputs = NodeHelpers.getNodeOutputs(workflow, targetNode, nodeType);
-		const mainOutputs = outputs.filter((output) =>
-			typeof output === 'string'
-				? output === NodeConnectionType.Main
-				: output.type === NodeConnectionType.Main,
-		);
+		const mainOutputs = NodeHelpers.getNodeOutputs(workflow, targetNode, nodeType)
+			.map((output) => (typeof output === 'string' ? { type: output } : output))
+			.filter((output) => output.type === NodeConnectionType.Main);
```

Comment:
Nit - Might be a little simpler to filter out errors here and check length below. 
```suggestion
			.filter((output) => output.type === NodeConnectionType.Main && output.category !== 'error');
```

As `(mainOutputs.length === 2 && mainOutputs[1]?.category === 'error');` might be unsafe if the order changes for whatever UX reason, and error outputs come first for example.

Reply:
Ah, I was trying to preserve the original logic, but that is much better indeed, thank you!